src/App/Fossa/Config/Analyze.hs

```diff
@@ -245,10 +245,10 @@ loadConfig ::
   ) =>
   AnalyzeCliOpts ->
   m (Maybe ConfigFile)
-loadConfig AnalyzeCliOpts{commons = CommonOpts{optConfig}} = do
-  -- FIXME: We eventually want to use the basedir to inform the config file root
-  configRelBase <- sendIO getCurrentDir
-  resolveConfigFile configRelBase optConfig
+loadConfig AnalyzeCliOpts{analyzeBaseDir, commons = CommonOpts{optConfig}} = do
+  cwd <- sendIO getCurrentDir
+  configBaseDir <- resolveDir cwd (toText analyzeBaseDir)
+  resolveConfigFile configBaseDir optConfig
```

Comment:
[Optional] I think the behavior you have in this PR is correct and what ought to be - but this could be a breaking change for some customers. Let's add debug log saying looking for config file xyz path, so diagnostics is easier - if this turns out to be a breaking change.

Reply:
This is definitely a breaking change, I remember seeing a bug about this in the early days of v2 config files.  I think we should consider falling back to the current directory for a config file if none is found at the basedir.

Reply:
I think I agree with Wes — we should try to fall back to looking in the current directory if we don't find a config file in the working directory.

I'm not sure if we'd build it that way if were were just implementing this, but I think it's the right change to make in order to avoid breaking customers

Reply:
I'm fine with that if @scruffystuffs  and @meghfossa are. Wes and Megh, what do you think?

Reply:
Breaking is breaking.  We're almost certainly going to see bugs filed about this.  I know for a fact that a small number of users are using the wrong behavior, there were bug reports about the config file when we first launched it.

I agree with the change, and I'm willing to make it, AND I'm willing to drop the fallback idea.  My focus here is being proactive about reducing the bug report burden on both on-duty and the post-sales team.

I think the bug report burden here will be small, so I'm OK with this one.